### PR TITLE
Refactor handling of default_quote

### PR DESCRIPTION
**Pull request by Bibo-Joshi**
Right now, `bot.defaults.quote` is passed through the json data when decoding an update for it to be stored in `Message.default_quote`. As @n5y pointed out offline, it would be cleaner to just use `Message.bot.defaults.quote`. This is, what this PR does.

To not break backwards compatibility, I left some optional `default_quote` args in the Webhook classes untouched and also kept `default_quote` as argument for `Message` (with a warning that it will override the bots settings). That's discussable though, as nobody should have actually used that …

Closes parts of #1797

### CODE REVIEW

**Comment by Bibo-Joshi** [comment]
Latests commits  

* Remove `Message.default_quote` all along, making it a breaking change. Hence, changed PR to the new v13 branch. Also, this way, unpickling is no problem anymore.
* Makes sure that a bots default values are un/pickled along with it.
  * upside: the unpickled bot of an unpickled `Message` still has the correct default `quote` setting
  * downside: the unpickled bot is not the same as the updaters one. If the defaults for the updater are changed, the unpickled bots don't change accordingly. Then again, this is not really new, i.e. the behaviour is the same for the token. maybe that should at least be documented somewhere …

**Comment by Bibo-Joshi** [comment]
We should postpone this after #1994

**Comment by Bibo-Joshi** [comment]
Also, we could/should raise a warning, if both `bot` and `defaults` are passed to `Updater`, as after this the defaults passed to the updater have no effect.

File: telegram/ext/updater.py [hunk 1]
```diff
@@ -116,6 +118,12 @@ def __init__(self,
                  dispatcher=None,
                  base_file_url=None):
 
+        if defaults and bot:
+            warnings.warn('Passing defaults to an Updater has no effect, if a Bot is passed, '
+                          'too. Pass it to the Bot instead.',
```
Comment (Poolitzer) [inline telegram/ext/updater.py:123]:
```suggestion
            warnings.warn('Passing defaults to an Updater has no effect when a Bot is passed'
                          'as well. Pass them to the Bot instead.',
```

**Comment by Poolitzer** [comment]
Here is a lot of changes which I reviewed in different PRs, like the refactor of the jobqueue, the api kwargs or the persistence part. Is that intended?

**Pull request merged**